File: app/src/main/java/kztproject/jp/splacounter/api/TodoistClient.java
package kztproject.jp.splacounter.api;

import javax.inject.Inject;

import io.reactivex.Single;
import kztproject.jp.splacounter.model.UserResponse;

public class TodoistClient {

    private final TodoistService service;

    @Inject
    public TodoistClient(TodoistService service) {
        this.service = service;
    }

    public Single<UserResponse> getUser(String token) {
        return service.getUser(token, "*", "[\"user\"]");
    }

}

File: app/src/main/java/kztproject/jp/splacounter/api/TodoistClient.kt
package kztproject.jp.splacounter.api

import io.reactivex.Single
import kztproject.jp.splacounter.model.UserResponse
import javax.inject.Inject

class TodoistClient @Inject
internal constructor(private val service: TodoistService) {

    fun getUser(token: String): Single<UserResponse> {
        return service.getUser(token, "*", "[\"user\"]")
    }

}
